lib/ex_doc/formatter/html.ex

```diff
@@ -146,18 +146,46 @@ defmodule ExDoc.Formatter.HTML do
     |> ExDoc.DocAST.highlight(language, opts)
   end
 
-  defp output_setup(build, config) do
+  def setup_output(config, cleanup, create, root \\ nil) do
+    root = root || config.output
+    safety_path = Path.join(root, ".ex_doc")
+
+    cond do
+      File.exists?(safety_path) and File.exists?(root) ->
+        cleanup.(root, config)
+
+      not File.exists?(root) ->
+        create.(root, config)
+
+      true ->
+        raise """
+        ex_doc cannot output to #{root}:
+        Directory already exists and is not managed by ex_doc.
+
+        Try giving an unexisting directory as `--output`.
+        """
```

Comment:
What do you think about this one?

```suggestion
      File.ls(root) == {:ok, []} ->
        :ok

      true ->
        IO.warn(
          "ExDoc is outputting to an existing directory. " <>
            "Beware documentation output may be mixed with other entries"
        )
```

Instead of raising, we warn, but only if the root directory is empty?